prov/efa/src/efa_shm.c

```diff
@@ -98,11 +98,13 @@ void efa_shm_info_create(const struct fi_info *app_info, struct fi_info **shm_in
 	int ret;
 	struct fi_info *shm_hints;
 
-	char *shm_provider;
-	if (efa_env.use_sm2) {
-		shm_provider = "sm2";
-	} else {
-		shm_provider = "shm";
+	if (strcmp(efa_env.intranode_provider, "shm") &&
```

Comment:
Check is case sensitive. Which might be OK but it contradicts your PR description.

Reply:
I didn't have any checks before... those were recently added.